### PR TITLE
refactor to use "id" as variable internally, instead of hash.

### PR DESCRIPTION
**Pull request by dominictarr**
To be generally consistent (ls returns {id,...}) also for functions that accept options, id is accepted, and defaulted from first, before key and hash.

This change is backwards compatible.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -205,23 +198,25 @@ var Blobs = module.exports = function (config) {
   return {
     get: function (opts) {
       if(isHash(opts)) {
-        if(isEmptyHash(hash)) return pull.empty()
```
Comment (arj03) [inline index.js:208]: This actually looks like a bug in the old code

File: index.js [hunk 2]
```diff
@@ -205,23 +198,25 @@ var Blobs = module.exports = function (config) {
...
+        if(isEmptyHash(opts)) return pull.empty()
         return Read(toPath(dir, opts))
       }
-      var hash = opts.key || opts.hash
-      if(!isHash(hash))
+      var id = opts.id || opts.key || opts.hash
```
Comment (arj03) [inline index.js:204]: Maybe refactor this into a function

**Comment by arj03** [comment]
Other than the small comment I think this looks good and improves the quality of the code.

**Review by christianbundy — APPROVED** [review]
LGTM, please merge.

**Pull request merged**